### PR TITLE
Js fix quadrature permutations

### LINKED ISSUE #1088: Wrong permutation of points in constructor of Quadrature.
Quadrature created from existing lower-dimensional subquadrature does not produce correct point coordinates. When used in FESideValues, the quadrature points differ from points in FEValues using the subquadrature.

### PR DESCRIPTION
**Pull request by jstebel**
Resolves #1088.

### CODE REVIEW

File: unit_tests/fem/quadrature_test.cpp [hunk 1]
```diff
@@ -85,36 +85,70 @@ TEST(Quadrature, midpoint){
 }
 
 
+/// Map lower dimensional quadrature to element quadrature to element quadrature
```
Comment (Paulie14) [inline unit_tests/fem/quadrature_test.cpp]: fix comment :)

**Comment by jbrezmorf** [comment]
Is this still related to the problem we arrive at during changes in the velocity calculation?

**Comment by jstebel** [comment]
No. The problem arises only in Elasticity.

Od: Jan Brezina
Odesláno:úterý 18. června 2019 15:02
Komu: flow123d/flow123d
Kopie: Jan Stebel; Assign
Předmět: Re: [flow123d/flow123d] Js fix quadrature permutations (#1089)

Is this still related to the problem we arrive at during changes in the velocity calculation?

File: src/quadrature/quadrature.hh [hunk 2]
```diff
@@ -224,7 +224,7 @@ Quadrature<dim>::Quadrature(const Quadrature<dim-1> &subq, unsigned int sid, uns
         
         //permute
         for (unsigned int i=0; i<RefElement<dim>::n_nodes_per_side; i++) {
```
Comment (jbrezmorf) [inline src/quadrature/quadrature.hh]: Why there is the upper bound 'n_nodes_per_side' while the 'p' and 'pp' vectors have size 'dim'? Possibly improve readability by introducing a common constant within this method.

Comment (jstebel) [reply src/quadrature/quadrature.hh]: For simplex it holds:

RefElement<dim>::n_nodes_per_side = dim

While side_permutations have dimension n_permutations x n_nodes_per_side, points of course have dimension dim. But I do not see which of these 2 labels for the same value should be preferred.

Comment (jstebel) [reply src/quadrature/quadrature.hh]: Added assert and explanation in the beginning of the constructor.

**Pull request merged**